### PR TITLE
docker: use CMake 3.29.6

### PR DESCRIPTION
**Pull request by KrisThielemans**

### CODE REVIEW

File: docker/build_essential-ubuntu.sh [hunk 1]
```diff
@@ -21,10 +20,10 @@ apt-get clean
...
 
 # CMake
-#curl -o cmake.tgz -L https://github.com/Kitware/CMake/releases/download/v3.25.1/cmake-3.25.1-linux-x86_64.tar.gz
-#tar xzf cmake.tgz && rm cmake.tgz
-#ln -s cmake-*x86_64 cmake || true
-#export PATH="$PWD/cmake/bin:$PATH"
+curl -o cmake.tgz -L https://github.com/Kitware/CMake/releases/download/v3.29.6/cmake-3.29.6-linux-x86_64.tar.gz
+tar xzf cmake.tgz && rm cmake.tgz
+ln -s cmake-*x86_64 cmake || true
+export PATH="$PWD/cmake/bin:$PATH"
```
Comment (casperdcl) [inline docker/build_essential-ubuntu.sh:26]: This won't persist for subsequent steps.

Perhaps untar into `/usr/local/bin` instead?

**Review by casperdcl — CHANGES_REQUESTED** [review]

File: docker/build_essential-ubuntu.sh (hunk 1 above)
Comment (KrisThielemans) [reply docker/build_essential-ubuntu.sh:26]: ah well. It used to work, but I believe you. Not sure if that link will be sufficient.

Of course, we also have https://github.com/SyneRBI/SIRF-SuperBuild/blob/d725d08f3d18308f4f14ab1b249e70b2acdbdc67/VirtualBox/scripts/INSTALL_CMake.sh

File: docker/build_essential-ubuntu.sh [hunk 2]
```diff
@@ -21,10 +20,10 @@ apt-get clean
 pushd $INSTALL_DIR
 
 # CMake
-#curl -o cmake.tgz -L https://github.com/Kitware/CMake/releases/download/v3.25.1/cmake-3.25.1-linux-x86_64.tar.gz
-#tar xzf cmake.tgz && rm cmake.tgz
-#ln -s cmake-*x86_64 cmake || true
-#export PATH="$PWD/cmake/bin:$PATH"
+curl -o cmake.tgz -L https://github.com/Kitware/CMake/releases/download/v3.29.6/cmake-3.29.6-linux-x86_64.tar.gz
+tar xzf cmake.tgz && rm cmake.tgz
```
Comment (casperdcl) [inline docker/build_essential-ubuntu.sh:24]:
```suggestion
mkdir -p /usr/local
pushd /usr/local
tar xzf $INSTALL_DIR/cmake.tgz --strip 1
popd
rm cmake.tgz
```

File: SuperBuild/External_Gadgetron.cmake [hunk 3]
```diff
@@ -88,13 +88,15 @@ endif()
     "Build Gadgetron MATLAB gadgets (not required for SIRF)" ${default_Gadgetron_BUILD_MATLAB_SUPPORT})
   option(Gadgetron_USE_MKL "Instruct Gadgetron to build linking to the MKL. The user must be able to install MKL on his own." OFF)
 
+  # disable Gadgetron GPU code
+  set (${proj}_USE_CUDA OFF)
+  # TODO enable it once bug fixed
```
[the remark this answers is not in the document]
Comment (paskino) [reply SuperBuild/External_Gadgetron.cmake:93]: I got a different bug though!

```
34 errors detected in the compilation of "/opt/SIRF-SuperBuild/sources/Gadgetron/toolboxes/nfft/gpu/cuGriddingConvolution.cu".```

[the remark this answers is not in the document]
Comment (paskino) [reply SuperBuild/External_Gadgetron.cmake:93]: Possibly it is better to refer to
- https://github.com/SyneRBI/SIRF-SuperBuild/issues/619

**Pull request merged**